torchtext/experimental/vectors.py

```diff
@@ -0,0 +1,211 @@
+import csv
+import io
+import os
+
+import torch
+from torch import Tensor
+import torch.nn as nn
+
+from torchtext.utils import (
+    download_from_url,
+    extract_archive
+)
+
+
+def fast_text(language="en", unk_tensor=None):
+    r"""Create a fast text Vectors object.
+
+    Args:
+        language (str): the language to use for FastText.
+        unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token
+
+    Returns:
+        Vectors: a Vectors object.
+
+    """
+    url = 'https://dl.fbaipublicfiles.com/fasttext/vectors-wiki/wiki.{}.vec'.format(language)
+
+    vectors_file_name = os.path.join('.data', os.path.basename(url) + '.pt')
+    saved_vectors = load_vectors(vectors_file_name)
+    if saved_vectors:
+        return saved_vectors
+
+    downloaded_file_path = download_from_url(url)
+    csv_file_name, _ext = os.path.splitext(downloaded_file_path)
+    csv_file_path = '{}.csv'.format(csv_file_name)
+
+    # skip if csv file already exists
+    if not os.path.exists(csv_file_path):
+        # format downloaded file into csv format expected by `vectors_from_file_object()`
+        with open(downloaded_file_path, 'r') as f1:
+            with open(csv_file_path, "w") as f2:
+                csvWriter = csv.writer(f2)
+                for line in f1:
+                    csvWriter.writerow(line.split(' ', 1))
+            f2.close()
+        f1.close()
+
+    file_obj = open(csv_file_path, 'r')
+    vectors_obj = vectors_from_file_object(file_obj)
+    save_vectors(vectors_obj, vectors_file_name)
+
+    return vectors_obj
+
+
+def glo_ve(name="840B", dim=300, unk_tensor=None):
+    r"""Create a GloVe Vectors object.
+
+    Args:
+        name (str): the language to use for FastText.
+        unk_tensor (Tensor): a 1d tensor representing the vector associated with an unknown token.
+
+    Returns:
+        Vectors: a Vectors object.
+
+    """
+    urls = {
+        '42B': 'http://nlp.stanford.edu/data/glove.42B.300d.zip',
+        '840B': 'http://nlp.stanford.edu/data/glove.840B.300d.zip',
+        'twitter.27B': 'http://nlp.stanford.edu/data/glove.twitter.27B.zip',
+        '6B': 'http://nlp.stanford.edu/data/glove.6B.zip',
+    }
+
+    url = urls[name]
+
+    vectors_file_name = os.path.join('.data', 'glove.{}.{}d.pt'.format(name, str(dim)))
+    saved_vectors = load_vectors(vectors_file_name)
+    if saved_vectors:
+        return saved_vectors
+
+    downloaded_file_path = download_from_url(url)
+    extracted_file_path = extract_archive(downloaded_file_path)[0]
+
+    csv_file_name, _ext = os.path.splitext(extracted_file_path)
+    csv_file_path = '{}.csv'.format(csv_file_name)
+
+    # skip if csv file already exists
+    if not os.path.exists(csv_file_path):
+        # format downloaded file into csv format expected by `vectors_from_file_object()`
+        with io.open(extracted_file_path, encoding="utf8") as f1:
```

Comment:
This is because we need to modify the downloaded file to be consistent with the file_like format that `vectors_from_file_object()` method is expecting. Do you have suggestions as to how else I could do this?

Reply:
I was thinking about downing and read the file to load tokens and vectors directly without `vectors_from_file_object` func.